marquez_airflow/dag.py

```diff
@@ -438,4 +484,18 @@ def _get_location(task):
             else:
                 return get_location(task.dag.fileloc)
         except Exception as e:
-            DAG.log.warning(f'Unable to fetch the location. {e}')
+            log.warning(f'Unable to fetch the location. {e}',
+                        exc_info=True)
+
+    def _get_marquez_run_id(self, ti, dagrun, kwargs):
+        log.debug(f"_get_marquez_run_id({ti}, {dagrun}, {kwargs})")
+        job_name = f'{ti.dag_id}.{ti.task_id}'
+        session = kwargs.get('session')
+        job_run_ids = self._job_id_mapping.pop(
+            JobIdMapping.make_key(job_name, dagrun.run_id), session)
+
+        log.info(f"job_run_ids: {job_run_ids}")
+
+        job_run_ids_array = jsonpickle.decode(string=job_run_ids)
```

Comment:
How we handle run IDs is part of the codebase that has a lower test coverage than others (mostly driven by integration test). I'd feel more comfortable if changes to how we store / retrieve run IDs was in a separate PR. I'm also wondering if encoding with `json.dumps()` (see [#L376](https://github.com/MarquezProject/marquez-airflow/blob/add-run-id/marquez_airflow/dag.py#L376)), then decoding with `jsonpickle.decode()` is an issue?

Reply:
sure. one PR at a time.